### PR TITLE
send a dummy protocol instead of an empty whitelist

### PR DESCRIPTION
**Pull request by aronatkins**
SockJS regards an empty `protocols_whitelist` option as an indication that all
protocols should be enabled. To avoid this behavior, pass a "dummy" protocol to
SockJS when we end up with an empty set of whitelisted protocols. This isn't
useful in the real world but helps validate behavior when all protocols end up
disabled through whatever technique.

For a bit more context, `protocols_whitelist` is interpreted not as an empty set of protocols but as an unspecified whitelist, causing SockJS to revert to attempting all protocols here:
https://github.com/sockjs/sockjs-client/blob/v0.3.4/lib/sockjs.js#L23-L26

This is eventually interpreted at:
https://github.com/sockjs/sockjs-client/blob/v0.3.4/lib/utils.js#L280

Note that this PR also changes how the protocol chooser is exposed to lib/sockjs.js to make it easier to stub in test.

### CODE REVIEW

**Review by jonyoder — COMMENTED** [review]
LGTM. 👍

File: lib/sockjs.js [hunk 1]
```diff
@@ -26,19 +25,27 @@ global.__shinyserverdebug__ = {
...
-exports.createFactory = function(options) {
+exports.createFactory = function(protocolChooser, options) {
   return function(url, context, callback) {
     if (!callback) throw new Error("callback is required");
 
     url = pathParams.reorderPathParams(url, ["n", "o", "t", "w", "s"]);
 
-    let whitelist = [];
-    require("./protocol-chooser").whitelist.forEach(prot => {
+    var whitelist = [];
```
Comment (jcheng5) [inline lib/sockjs.js:34]: Just curious, why `let` became `var` here?

Comment (aronatkins) [reply lib/sockjs.js:34]: Copy-paste mistake. I originally patched the generated `shiny-server-client.js` in Connect and then moved code from there to here. Fixing now.

Comment (aronatkins) [reply lib/sockjs.js:34]: Reverted with #9

Comment (jcheng5) [reply lib/sockjs.js:34]: Oh that's anti climactic, I thought I was about to learn something new about ES6 :)

**Pull request merged**